### DISCUSSION TITLE
workflow-controller is failing health check

### DISCUSSION THREAD

**Discussion by minty99** [argoproj/argo-workflows#6176] [Q&A] [answered]
```
~/Work
❯ kubectl get pods -l app=workflow-controller  --all-namespaces
NAMESPACE   NAME                                   READY   STATUS    RESTARTS   AGE
argo        workflow-controller-798fbf87f6-72wg6   1/1     Running   4900       21d
```
```
Back-off restarting failed container
Liveness probe failed: Get http://10.240.3.106:6060/healthz: dial tcp 10.240.3.106:6060: connect: connection refused
```
workflow-controller deployment is failing health check, and it restarts every 10~15 minutes. How can I fix this error?
There is no problem to run workflows, but I suffer by many health check failure notifications.

**Comment by alexec** [comment, marked as answer]
There was a bug in one version of v3.0.x. Please make sure you’re running >= v3.0.7

**Reply by minty99** [reply]
@alexec Thanks! The issue is fixed by upgrading from v3.0.5 to v3.0.7.
